marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/calculateMergingActivityLib.sjs

```diff
@@ -0,0 +1,54 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+const currentDatabaseName = xdmp.databaseName(xdmp.database());
+
+function calculateMergingActivity(step)
+{
+    let sourceDatabaseName = step.sourceDatabase || currentDatabaseName;
+    if (sourceDatabaseName !== currentDatabaseName) {
+        const HubUtils = require("/data-hub/5/impl/hub-utils.sjs");
+        return new HubUtils().queryLatest(function () {
```

Comment:
This is a bit tangential, but what is "queryLatest" supposed to mean? It's more confusing to see that than "xdmp.invokeFunction" - I have to lookup queryLatest to see what it does, and the only thing of note is that it passes ignoreAmps:true in - and there's no indication of why it's doing that. So why use queryLatest instead of xdmp.invokeFunction?

Reply:
It has been used to query at the latest timestamp. At the time the function was created we were trying to consolidate functions, like invoke/eval, that required special privileges.